iftk/helpers/deepgram_stream_queue.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import asyncio
+import json
+import pyaudio
+import websockets
+
+FORMAT = pyaudio.paInt16
+CHANNELS = 1
+RATE = 44100
+CHUNK = 512
+DEEPGRAM_CHUNK = 8000
+DEEPGRAM_RATE = 16000
+SECONDS = 2
+WSS_URL = "wss://api.deepgram.com/v1/listen?endpointing=500&encoding=linear16&sample_rate=16000&channels=1&interim_results=false"
+
+
+async def DeepgramStream(key: str, transcription_queue: asyncio.Queue) -> None:
+    """Put ASR results from Deepgram streaming API in an asyncio.Queue.
+
+    Args:
+        key (str): Your Deepgram API key
+        transcription_queue (asyncio.Queue): A queue that stores the transcriptions as a
+        "speech_final" message is detected from Deepgram.
+    """
+    extra_headers = {"Authorization": f"Token {key}"}
+    audio_queue = asyncio.Queue()
+
+    def callback(in_data, frame_count, time_info, status):
+        """A basic callback for Deepgram's stream."""
+        audio_queue.put_nowait(in_data)
+        return (None, pyaudio.paContinue)
+
```

Comment:
Because asyncio.Queue is not thread safe, and deepgram calls back on some thread, we'll need to put this back on the asyncio loop.

```suggestion
    loop = asyncio.get_running_loop()
    
    def callback(in_data, frame_count, time_info, status):
        """A basic callback for Deepgram's stream."""
        loop.call_soon_threadsafe(audio_queue.put_nowait, in_data)
        return (None, pyaudio.paContinue)

```